gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs-settings/api-runtime-logs-settings.component.ts

```diff
@@ -0,0 +1,242 @@
+        })
+      : new FormGroup({
+          enabled: this.enabledFormControl,
+        });
+    this.form.updateValueAndValidity();
+  }
+
+  private handleEnabledChanges(): void {
+    this.form
+      .get('enabled')
+      .valueChanges.pipe(takeUntil(this.unsubscribe$))
+      .subscribe((value) => {
+        this.enabled = value;
+        this.samplingType = this.enabled ? 'PROBABILITY' : undefined;
+        this.initForm();
+        this.handleSamplingTypeChanges();
+        this.handleLoggingModeChanges();
+      });
+  }
+
+  private handleSamplingTypeChanges(): void {
+    if (this.enabled) {
+      this.samplingTypeSubscription$ = this.form
+        .get('samplingType')
+        .valueChanges.pipe(takeUntil(this.unsubscribe$))
+        .subscribe((value) => {
+          this.samplingType = value;
+          const samplingValueControl = this.form.get('samplingValue');
+          samplingValueControl.setValue(this.getSamplingDefaultValue(value));
+          samplingValueControl.setValidators(this.getSamplingValueValidators(value));
+          samplingValueControl.updateValueAndValidity();
+        });
+    } else {
+      this.samplingTypeSubscription$?.unsubscribe();
+    }
+  }
+
+  private handleLoggingModeChanges(): void {
+    if (this.enabled) {
+      this.loggingModeSubscription$ = merge(this.form.get('entrypoint').valueChanges, this.form.get('endpoint').valueChanges)
+        .pipe(takeUntil(this.unsubscribe$))
+        .subscribe(() => {
+          const formValues = this.form.getRawValue();
+          this.hasLoggingModeEnabled = formValues.entrypoint || formValues.endpoint;
+          if (this.hasLoggingModeEnabled) {
+            this.form.get('messageContent').enable();
+            this.form.get('messageHeaders').enable();
+            this.form.get('messageMetadata').enable();
+            this.disableAndUncheck('requestPayload');
+            this.disableAndUncheck('requestHeaders');
+          } else {
+            this.disableAndUncheck('messageContent');
+            this.disableAndUncheck('messageHeaders');
+            this.disableAndUncheck('messageMetadata');
+            this.form.get('requestPayload').enable();
+            this.form.get('requestHeaders').enable();
+          }
+        });
+    } else {
+      this.loggingModeSubscription$?.unsubscribe();
+    }
+  }
+
+  private disableAndUncheck(controlName: string): void {
+    this.form.get(controlName).setValue(false);
+    this.form.get(controlName).disable();
+  }
+
+  private getSamplingValueValidators(samplingType: SamplingTypeEnum): ValidatorFn[] {
+    switch (samplingType) {
+      case 'PROBABILITY':
+        return [Validators.required, Validators.min(0.01), Validators.max(0.5)];
+      case 'COUNT':
+        return [Validators.required, Validators.min(0)];
+      case 'TEMPORAL':
+        return [Validators.required];
```

Comment:
**suggestion (non-blocking)**:
Should we use something like this: https://www.npmjs.com/package/iso8601-duration
to validate the format ?

Reply:
It was a question I wanted to raise during our daily meeting 😅